activerecord/lib/active_record/signed_id.rb

```diff
@@ -70,10 +70,11 @@ def find_signed!(signed_id, purpose: nil)
       # Rails.application.key_generator. By default, it's SHA256 for the digest and JSON for the serialization.
       def signed_id_verifier
         @signed_id_verifier ||= begin
-          if signed_id_verifier_secret.nil?
-            raise ArgumentError, "You must set ActiveRecord::Base.signed_id_verifier_secret to use signed ids"
+          if secret = signed_id_verifier_secret
+            secret = secret.call if secret.respond_to?(:call)
```

Comment:
the result of `secret.call` can also be `nil`, should we also raise the exception on that case?

Reply:
Yes, good call